GREYErrorFormatter.m

```diff
@@ -0,0 +1,259 @@
+                                                       space:NO
+                                               secondNewLine:YES
+                                                 secondEntry:searchActionInfo]];
+    }
+  }
+  
+  if ([keys containsObject:kErrorFormatterScreenshotPathsKey]) {
+    for (NSString *key in _error.appScreenshots.allKeys) {
+      NSString *screenshotPath = [NSString stringWithFormat:@"%@", _error.appScreenshots[key]];
+      [logger addObject:[self errorLogEntryWithNewlineBefore:NO
+                                                       entry:key
+                                                       colon:YES
+                                                       space:YES
+                                               secondNewLine:NO
+                                                 secondEntry:screenshotPath]];
+    }
+  }
+  
+  if ([keys containsObject:kErrorFormatterUnderlyingErrorKey]) {
+    NSString *nestedError = _error.nestedError.description;
+    if (nestedError) {
+      [logger addObject:[self errorLogEntryWithNewlineBefore:NO
+                                                       entry:kErrorFormatterUnderlyingErrorKey
+                                                       colon:YES
+                                                       space:NO
+                                               secondNewLine:YES
+                                                 secondEntry:nestedError]];
+    }
+  }
+  
+  if ([keys containsObject:kErrorFormatterUIHierarchyKey]) {
+    NSString *UIHierarchy = GREYFormattedHierarchy(_error.appUIHierarchy);
+    if (UIHierarchy) {
+      [logger addObject:UIHierarchy];
+    }
+  }
+  
+  if ([keys containsObject:kErrorFormatterStackTraceKey]) {
+    NSArray<NSString *> *stackTrace = _error.stackTrace;
+    if (stackTrace) {
+      [logger addObject:[self errorLogEntryWithNewlineBefore:NO
+                                                       entry:kErrorStackTraceKey
+                                                       colon:YES
+                                                       space:YES
+                                               secondNewLine:NO
+                                                 secondEntry:stackTrace.description]];
+    }
+  }
+  
+  return [logger componentsJoinedByString:@"\n"];
+}
+
+// Example: errorLogEntryWithNewlineBefore:NO entry:@"1" colon:YES space:YES
+// secondNewLine:YES secondEntry:@"2" -> @"1: \n2\n"
+// Notice it always adds a \n to the end, so there is no need to supply that as a parameter.
+- (NSString *)errorLogEntryWithNewlineBefore:(BOOL)lineBefore
+                                       entry:(NSString *)entry
+                                       colon:(BOOL)colon
+                                       space:(BOOL)space
+                               secondNewLine:(BOOL)secondNewLine
+                                 secondEntry:(NSString *)secondEntry {
```

Comment:
Won't this be expanded ad nauseam in case newer keys are to be added? I think it would be better to have the NSString contain the spacing and newline info instead.